internal/core/delta.go

```diff
@@ -10,27 +10,9 @@
 
 package core
 
-import (
-	cid "github.com/ipfs/go-cid"
-)
-
 // Delta represents a delta-state update to delta-CRDT.
 // They are serialized to and from Protobuf (or CBOR).
 type Delta interface {
```

Comment:
I have a preference to have two different interfaces for this, `Delta`, and whatever `IPLDSchemaBytes` is defined on.  They represent two different concepts that happen to be implemented by the same concrete types atm.  If we really want them on one interface, that interface should embed `Delta` and `IPLDSchemaBytes`'s host-interface.

Reply:
I prefer to keep them separate. Like Andy said, they represent two different concepts.  It is much better in my opinion to keep interfaces as small as possible. We already suffer from bloated interfaces or at least using large interfaces too liberally.

Reply:
The only problem here is that unless we add more compile time interface to checks to the deltas, we lose out on the compile time safety.

It's not the biggest issue, but this change now requires that deltas have a corresponding IPLD schema.

We do still preserve some compile time safety, since if you pass the delta into the mustSchemas func it'll complain. 

But there's no clear statement in the code that deltas much have IPLD schemas, and if it's a contract we are requiring, that contract should be expressed in the code, ideally with compile time enforcement